aws_analysis_tools/cli/update_ec2_tags.py

```diff
@@ -43,120 +41,134 @@ def add_cli_arguments(self, parser):
-                    'Could not determine instance_id (%s) and/or ec2_region (%s)' % \
-                    (instance_id, ec2_region)
+                    'Could not determine instance_id ({instance_id}) and/or ec2_region ({region})'.format(
+                        instance_id=instance_id,
+                        region=ec2_region,
+                    )
                 )
 
-            ### many places the info can come from, so let us know here what we found
+            # many places the info can come from, so let us know here what we found
             log.info(
                 'Proceeding with instance id: %s - region: %s - classes: %s',
                 instance_id, ec2_region, ' '.join(classes)
             )
 
-            ###
-            ### Set up the tags we want to use
-            ###
+            #
+            # Set up the tags we want to use
+            #
             tags_dict = {
-                'environment':  environment,
+                'environment': environment,
                 'cluster_name': cluster_name,
-                EC2_CLASS_KEY:  ",".join(classes),
+                's_classes': ",".join(classes),
+                'release': release,
             }
 
-            ### quick dump of what we're about to do send.
-            for k,v in tags_dict.iteritems():
+            # quick dump of what we're about to do send.
+            for k, v in tags_dict.iteritems():
                 log.info('Setting tag "%s" to: %s', k, v)
 
-
-            ###
-            ### Now do the actual update, if desired
-            ###
+            #
+            # Now do the actual update, if desired
+            #
             if dry_run:
                 log.info('Dry run - not updating instance in EC2')
 
             else:
                 log.info('Updating instance %s in region %s', instance_id, ec2_region)
-                region_obj  = self.boto.ec2.get_region(ec2_region)
-                ec2         = self.boto.connect_ec2(region = region_obj)
+                region_obj = self.boto.ec2.get_region(ec2_region)
+                ec2 = self.boto.connect_ec2(region=region_obj)
 
-                ### ec2 calls throw exceptions when they fail
+                # ec2 calls throw exceptions when they fail
                 try:
                     ec2.create_tags([instance_id], tags_dict)
                     stats.incr('ec2_tag_update')
                     log.info('Update completed successfully')
 
-                ### no matter what happens, we want to catch it and make sure
-                ### log it appropriately before we re-raise
-                except Exception, e:
+                # no matter what happens, we want to catch it and make sure
+                # log it appropriately before we re-raise
+                except Exception:
                     stats.incr('error.ec2_tag_update')
-                    self.raise_critical_error('Tag update failed: %s' % e)
+                    raise
```

Comment:
`self.raise_critical_error()` has handled within [`app.context()`](https://github.com/krux/python-krux-stdlib/blob/master/krux/cli.py#L257-L283) context manager.

I was able to get a proper stack trace with blank `raise`. Let me test it in other instances.